packages/form-plugin/src/form.plugin.ts

```diff
@@ -22,9 +22,9 @@ export class NgxsFormPlugin implements NgxsPlugin {
     let nextState = state;
 
     if (type === UpdateFormValue.type || type === UpdateForm.type) {
-      nextState = setValue(nextState, `${event.payload.path}.model`, {
-        ...event.payload.value
-      });
+      const payloadValue = Array.isArray(event.payload.value) ? [...event.payload.value] : { ...event.payload.value };
```

Comment:
Refactor

```ts
const value = event.payload.value;
const payloadValue = Array.isArray(value) ? [ ...value ] : { ...value };
```

Reply:
The es6 style is preferred:
`const { value } = event.payload;`

Reply:
done